src/student_success_tool/analysis/pdp/targets/failure_to_graduate_within_x_years.py

```diff
@@ -0,0 +1,142 @@
+import logging
+import typing as t
+from collections.abc import Collection
+import pandas as pd
+from student_success_tool.analysis.pdp.targets import shared
+from student_success_tool.analysis.pdp import utils
+
+LOGGER = logging.getLogger(__name__)
+
+
+def make_labeled_dataset(
+    df: pd.DataFrame,
+    *,
+    student_criteria: dict[str, object | Collection[object]],
+    student_id_cols: str | list[str] = "student_guid",
+    exclude_pre_cohort_terms: bool = True,
+    term_is_pre_cohort_col: str = "term_is_pre_cohort",
+    years_to_bach_at_cohort_col: str = "first_year_to_bachelors_at_cohort_inst",
+    max_years_to_graduation: int = 4,
+    intensity_time_lefts: list[tuple[str, float, t.Literal["year", "term"]]],
+    max_term_rank: int,
+    num_terms_in_year: int = 4,
+    enrollment_intensity_col: str = "enrollment_intensity_first_term",
+    term_rank_col: str = "term_rank",
+) -> pd.DataFrame:
+    """
+    Make a labeled dataset for modeling, where each row consists of features
+    from eligible students' first term (within their cohort)
+    matched to computed target variables.
+
+    See Also:
+        - :func:`select_eligible_students()`
+        - :func:`compute_target_variable()`
+        - :func:`shared.get_first_student_terms_within_cohort()`
+    """
+    df_eligible_students = select_eligible_students(
+        df,
+        student_criteria=student_criteria,
+        student_id_cols=student_id_cols,
+        intensity_time_lefts=intensity_time_lefts,
+        max_term_rank=max_term_rank,
+        num_terms_in_year=num_terms_in_year,
+        enrollment_intensity_col=enrollment_intensity_col,
+        term_rank_col=term_rank_col,
+    )
+    df_eligible_student_terms = pd.merge(
+        df, df_eligible_students, on=student_id_cols, how="inner"
+    )
+    if exclude_pre_cohort_terms is True:
+        df_features = shared.get_first_student_year_term_within_cohort(
+            df_eligible_student_terms,
+            student_id_cols=student_id_cols,
+            term_is_pre_cohort_col=term_is_pre_cohort_col,
+            sort_cols=term_rank_col,
+            include_cols=None,
+        )
+    else:
+        df_features = shared.get_first_student_year_term(
+            df_eligible_student_terms,
+            student_id_cols=student_id_cols,
+            sort_cols=term_rank_col,
+            include_cols=None,
+        )
+    df_targets = compute_target_variable(
+        df_eligible_student_terms,
+        student_id_cols=student_id_cols,
+        years_to_bach_at_cohort_col=years_to_bach_at_cohort_col,
+    )
+    df_labeled = pd.merge(df_features, df_targets, on=student_id_cols, how="inner")
+    return df_labeled
+
+
+def compute_target_variable(  
+    df: pd.DataFrame,  
+    *,  
+    student_id_cols: str | list[str] = "student_guid",  
+    years_to_bach_at_cohort_col: str = "first_year_to_bachelors_at_cohort_inst",  
+    intensity_time_lefts: list[tuple[str, float, t.Literal["year", "term"]]], 
```

Comment:
It looks like you never use this argument in the function's logic. Why's it here then?